signingscript/src/signingscript/task.py

```diff
@@ -301,6 +316,13 @@ def build_filelist_dict(context):
                 "full_path": full_path,
                 "formats": _sort_formats(artifact_dict["formats"]),
             }
+            if authenticode_comment and 'msi' in path:
+                if isinstance(authenticode_comment, type({})):
+                    if path in authenticode_comment:
+                        filelist_dict[path]["comment"] = authenticode_comment[path]
+                else:
+                        filelist_dict[path]["comment"] = authenticode_comment
+
```

Comment:
We're getting to a level of complexity where we may want to call a helper function to build `filelist_dict[path]`.